GNU patch accepts the --- and +++ lines in any order.
+++ filename
--- filename
@@ -5,7 +5,7 @@ foo
 aaa
 bbb
 ccc
-ddd
+eee
 fff
 ggg
 hhh
